### PR TITLE
Add BYTE_STREAM_SPLIT data file

### PR DESCRIPTION
**Pull request by mwlon**
Allow better testing of BYTE_STREAM_SPLIT encoding implementations.

### CODE REVIEW

**Comment by mwlon** [comment]
@tustvold This should help with the byte stream split PR I'll make soon

**Comment by mapleFU** [comment]
Should we link `byte_stream_split.zstd.parquet` in readme.md?

**Comment by mwlon** [comment]
Yep, added it. I hadn't noticed the readme.

File: data/byte_stream_split.zstd.parquet.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+`byte_stream_split.zstd.parquet` is generated by pyarrow 14.0.2 using the following code:
```
Comment (pitrou) [inline data/byte_stream_split.zstd.parquet.md:1]: Perhaps inline this in the README as for other files?

File: data/byte_stream_split.zstd.parquet.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+from pyarrow import parquet as pq
+import numpy as np
+
+np.random.seed(0)
+table = pa.Table.from_pydict({
+  'f32': np.random.normal(size=300).astype(np.float32),
+  'f64': np.random.normal(size=300).astype(np.float64),
+})
+
+f = '/Users/martin/Downloads/byte_stream_split.parquet'
```
Comment (pitrou) [inline data/byte_stream_split.zstd.parquet.md:14]: Just writing into the current directory would make this script more readily usable.

Comment (mwlon) [reply data/byte_stream_split.zstd.parquet.md]: Oops - fixed

File: data/README.md [hunk 3]
```diff
@@ -49,6 +49,7 @@
 | float16_nonzeros_and_nans.parquet | Float16 (logical type) column with NaNs and nonzero finite min/max values |
 | float16_zeros_and_nans.parquet    | Float16 (logical type) column with NaNs and zeros as min/max values. . See [note](#float16-files) below |
 | concatenated_gzip_members.parquet     | 513 UINT64 numbers compressed using 2 concatenated gzip members in a single data page |
+| byte_stream_split.zstd.parquet | standard normals with `BYTE_STREAM_SPLIT` encoding |
```
Comment (pitrou) [inline data/README.md:52]:
```suggestion
| byte_stream_split.zstd.parquet | Standard normals with `BYTE_STREAM_SPLIT` encoding. See [note](#byte-stream-split) below |
```

File: data/README.md [hunk 4]
```diff
@@ -321,3 +322,32 @@ print(m2.row_group(0).column(0))
 #   total_compressed_size: 76
 #   total_uncompressed_size: 76
 ```
+
+# Byte Stream Split
```
Comment (pitrou) [inline data/README.md:326]:
```suggestion
## Byte Stream Split
```

**Pull request merged**